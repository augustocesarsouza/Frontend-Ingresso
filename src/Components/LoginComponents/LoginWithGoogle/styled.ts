import styled from 'styled-components';

export const ContainerLoginWithGoogleMain = styled.div`
  display: flex;
  width: 80%;
  justify-content: flex-start;
  border-bottom: 1px solid #e2e2e273;
  padding: 14px 25px;
`

export const ContainerLoginWithGoogle = styled.div`
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 32rem;
`

export const H1 = styled.h1`
  font-size: 21px;
  color: rgb(102, 102, 102);
`

export const ContainerGoogle = styled.div`
  background: rgb(66, 133, 244);
  padding: 9px 7px;
  display: flex;
  align-items: center;
`

export const ContainerSvgGoogle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Span = styled.span`
  margin-left: 15px;
  color: white;
`