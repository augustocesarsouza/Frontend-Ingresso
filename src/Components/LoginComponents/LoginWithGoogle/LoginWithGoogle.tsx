import GoogleSvg from '../../../Svg/GoogleSvg';
import * as Styled from './styled';

const LoginWithGoogle = () => {
  return (
    <Styled.ContainerLoginWithGoogleMain>
      <Styled.ContainerLoginWithGoogle>
        <Styled.H1>Entrar com uma Rede Social</Styled.H1>
        <Styled.ContainerGoogle>
          <Styled.ContainerSvgGoogle>
            <GoogleSvg />
          </Styled.ContainerSvgGoogle>
          <Styled.Span>Entrar com o Google</Styled.Span>
        </Styled.ContainerGoogle>
      </Styled.ContainerLoginWithGoogle>
    </Styled.ContainerLoginWithGoogleMain>
  );
};

export default LoginWithGoogle;
