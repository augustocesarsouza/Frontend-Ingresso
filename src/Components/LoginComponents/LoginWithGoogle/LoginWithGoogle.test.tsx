import { render, screen } from '@testing-library/react';
import LoginWithGoogle from './LoginWithGoogle';

describe('LoginWithGoogle.test', () => {
  test('should render header svgGoogle, and span', () => {
    const { container } = render(<LoginWithGoogle />);
    const h1 = screen.getByRole('heading');
    expect(h1).toBeInTheDocument();

    const svgGoogle = container.querySelector('.svg-google');
    expect(svgGoogle).toBeInTheDocument();

    const spanEnter = screen.getByText('Entrar com o Google');
    expect(spanEnter).toBeInTheDocument();

    expect(container).toMatchSnapshot();
    expect.assertions(4);
  });
});
