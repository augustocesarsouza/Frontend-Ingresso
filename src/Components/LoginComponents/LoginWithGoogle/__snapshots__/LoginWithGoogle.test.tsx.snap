// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithGoogle.test should render header svgGoogle, and span 1`] = `
.c0 {
  display: flex;
  width: 80%;
  justify-content: flex-start;
  border-bottom: 1px solid #e2e2e273;
  padding: 14px 25px;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 32rem;
}

.c2 {
  font-size: 21px;
  color: rgb(102, 102, 102);
}

.c3 {
  background: rgb(66, 133, 244);
  padding: 9px 7px;
  display: flex;
  align-items: center;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
}

.c5 {
  margin-left: 15px;
  color: white;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Entrar com uma Rede Social
      </h1>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            class="svg-google"
            height="26"
            role="img"
            viewBox="0 0 23 23"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <rect
                fill="#FFF"
                height="26"
                rx="1"
                width="26"
              />
              <path
                d="M3.99 7.956c.374-.806.903-1.51 1.524-2.146 1.404-1.44 3.104-2.363 5.117-2.677 2.817-.44 5.338.21 7.516 2.045.138.116.172.184.022.327-.769.737-1.526 1.486-2.287 2.23-.078.077-.13.17-.27.046-1.917-1.72-5.048-1.7-7.086.154a5.72 5.72 0 0 0-1.524 2.275c-.047-.03-.097-.058-.142-.091L3.99 7.956"
                fill="#D7282A"
              />
              <path
                d="M6.98 13.723c.275.683.616 1.329 1.124 1.878 1.291 1.395 2.892 1.997 4.81 1.802.89-.09 1.702-.386 2.461-.84.073.064.142.132.219.191.887.678 1.776 1.355 2.665 2.032a7.68 7.68 0 0 1-3.438 1.839c-3.068.759-5.897.28-8.408-1.66A8.377 8.377 0 0 1 3.982 16l2.998-2.277"
                fill="#45AC43"
              />
              <path
                d="M18.259 18.786c-.889-.677-1.778-1.354-2.665-2.032-.077-.059-.146-.127-.219-.19.602-.45 1.104-.98 1.433-1.656a5.72 5.72 0 0 0 .312-.836c.06-.196.041-.273-.207-.27-1.48.012-2.958.005-4.437.005-.313 0-.313 0-.313-.317 0-.98.004-1.962-.005-2.942-.001-.19.032-.262.25-.262 2.728.008 5.456.006 8.183.003.148 0 .24.01.266.186.34 2.334.067 4.565-1.181 6.625-.383.63-.842 1.207-1.417 1.686"
                fill="#4485F4"
              />
              <path
                d="M6.98 13.723 3.982 16c-.488-.888-.77-1.841-.9-2.835a8.707 8.707 0 0 1 .768-4.957c.04-.086.093-.168.14-.252l2.87 2.163c.045.033.095.06.142.091-.4 1.169-.38 2.34-.022 3.513"
                fill="#F4C300"
              />
            </g>
          </svg>
        </div>
        <span
          class="c5"
        >
          Entrar com o Google
        </span>
      </div>
    </div>
  </div>
</div>
`;
